tutorials/data-science-extraction/index.md

```diff
@@ -0,0 +1,390 @@
+[identify faces, landmarks, and text][vision] in images; [translate][translate]
+between languages; and [provide structure to prose][natural-language].
+
+In this tutorial, you'll write several functions to perform various
+transformations and extraction to turn raw audio files into structured,
+queryable data. These functions can then be easily combined together into a
+reusable data ingestion pipeline, as described in the
+[preprocessing](preprocessing.md) tutorial.
+
+[speech]: /speech
+[vision]: /vision
+[translate]: /translate
+[natural-language]: /natural-language
+
+### Objectives
+
+This data extraction pipeline example can be described as a series of discrete
+steps:
+
+* Unzipping a collection of `mp3` audio files
+* Converting the `mp3`s to raw `LINEAR16` audio
+* Uploading the raw audio files to Google Cloud Storage
+* Extract transcription of speech from the audio files
+* Process the transcription to identify the entities.
+* Save the results
+
+### Prerequisites
+
+* You've enabled the [Cloud Speech API][speech-api] and the [Cloud Natural
+  Language API][nl-api] in your Cloud Console project.
+* You've downloaded [service account credentials][service-account] and set the
+  `GOOGLE_APPLICATION_DEFAULT` environment variable to point to them. ([More
+  details][auth])
+* You have basic familiarity with [Python][python] programming.
+* You've installed [pip][pip].
+* You've installed [virtualenv][virtualenv]
+
+[speech-api]: https://console.cloud.google.com/apis/api/speech.googleapis.com/overview?project=_
+[nl-api]: https://console.cloud.google.com/apis/api/language.googleapis.com/overview?project=_
+[service-account]: https://console.cloud.google.com/apis/credentials?project=_
+[python]: https://www.python.org/
+[pip]: https://pip.pypa.io/en/latest/installing/
+[virtualenv]: http://virtualenv.pypa.io
+[auth]: /docs/authentication#developer_workflow
+
+## Download the tutorial data
+
+For this tutorial, we'll extract data from readings of Aesop's Fables from
+[LibriVox](//librivox.org/author/181) for demonstration purposes. For
+convenience, we've cached a copy of the zip files in a [Google Cloud
+Storage][storage] bucket:
+
+* [Download][aesop1.zip] the first audio file, to use while writing and testing
+  your preprocessing functions.
+* You can also browse the whole collection of audio files [here][aesop-gcs].
+
+[aesop1.zip]: https://storage.cloud.google.com/data-science-getting-started/aesop_fables_volume_one_librivox_64kb_mp3.zip
+[aesop-gcs]: https://console.cloud.google.com/storage/browser/data-science-getting-started/?project=_
+[storage]: /storage
+
+## Convert the file
+
+The file as downloaded from LibriVox is in a zip archive, which we'll need to
+convert into a format that the API accepts. In order to do this, we'll first
+define a function that unzips the archive, producing each file successively:
+
+[embedmd]:# (src/extraction/unzip.py /def unzip/ /^$/)
+```py
+def unzip(filename):
```

Comment:
Hm... good point. I added a link to the complete file right before the "now run this" instruction. I realize I need this for the community site anyway, and it should also provide this kind of boilerplate without having to interrupt the narrative.